src/parser/jobs/pld/modules/Confiteor.tsx

```diff
@@ -0,0 +1,73 @@
+import {Plural, Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+export class Confiteor extends Analyser {
+	static override handle = 'confiteor'
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+
+	private droppedConfiteors: number = 0
+	private usedConfiteor: boolean = false
+
+	private confiteorHook?: EventHook<Events['action']>
+
+	override initialise() {
+		this.addEventHook({
+			type: 'statusApply',
+			source: this.parser.actor.id,
+			status: this.data.statuses.CONFITEOR_READY.id,
+		}, this.onGain)
+
+		this.addEventHook({
+			type: 'statusRemove',
+			source: this.parser.actor.id,
+			status: this.data.statuses.CONFITEOR_READY.id,
+		}, this.onDrop)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGain() {
+		this.confiteorHook = this.addEventHook(
+			filter<Event>()
+				.source(this.parser.actor.id)
+				.type('action')
+				.action(this.data.actions.CONFITEOR.id),
+			() => this.usedConfiteor = true,
+		)
+	}
+
+	private onDrop() {
+		if (!this.usedConfiteor) { this.droppedConfiteors++ }
+		if (this.confiteorHook != null) {
+			this.removeEventHook(this.confiteorHook)
+			this.confiteorHook = undefined
+		}
+		this.usedConfiteor = false
+	}
+
+	private onComplete() {
+		this.suggestions.add(new TieredSuggestion({
```

Comment:
does this suggestion add anything that the requi required actions one does not? same link i logged on the other one trips this one as well, and it really feels like it's repeating the same thing.

Reply:
I personally think this is specific and major enough to deserve it's own call out. Letting the buff time out is HUGE imo.

Reply:
oh, huh, confetti ready is a 30s buff, so it's entirely possible to hold it and use outside reqi. whack. _that said_, it's only actually a particularly strong skill with reqi, each skill loses 500 (fucking hell that's a lot) potency without it. so it's sorta a two-way relation - confetti combo is basically shit outside reqi, and reqi should be wholly consumed on confetti.

so... yeah, i _think_ it's still dupey, but perhaps a combination will require rethinking at least the wording to be very clear that _"reqi is for confetti combo"_, rather than "you want to get these in i guess". perhaps we can, like, consider the 4-hit combo as a single "thing" to do in reqi, and _any_ drop of those as a full-failure, rather than (i.e.) HS->combo just considering the last hit dropped.

just thoughts.

Reply:
I definitely agree that requi is for confetti only, and if you do let it time out, there's no way that the requi module will be happy (because it won't have the confetti spells). I can definitely see the possibility of the duplication making a bigger deal about it than it should be. I'm just struggling with the idea of there being a use case that isn't getting addressed directly (although it is indirectly in the requi module).